pkgs/applications/science/electronics/kicad/default.nix

```diff
@@ -1,52 +1,165 @@
 
 with lib;
 let
-
   baseName = if (stable) then "kicad" else "kicad-unstable";
+  versionsImport = import ./versions.nix;
 
-  versions =  import ./versions.nix;
-  versionConfig = versions.${baseName};
+  # versions.nix does not provide us with version, src and rev. We
+  # need to turn this into approprate fetcher calls.
+  kicadSrcFetch = fetchFromGitLab {
+    group = "kicad";
+    owner = "code";
+    repo = "kicad";
+    rev = versionsImport.${baseName}.kicadVersion.src.rev;
+    sha256 = versionsImport.${baseName}.kicadVersion.src.sha256;
+  };
 
-  wxGTK = if (stable)
+  i18nSrcFetch = fetchFromGitLab {
+    group = "kicad";
+    owner = "code";
+    repo = "kicad-i18n";
+    rev = versionsImport.${baseName}.libVersion.libSources.i18n.rev;
+    sha256 = versionsImport.${baseName}.libVersion.libSources.i18n.sha256;
+  };
+
+  libSrcFetch = name: fetchFromGitLab {
+    group = "kicad";
+    owner = "libraries";
+    repo = "kicad-${name}";
+    rev = versionsImport.${baseName}.libVersion.libSources.${name}.rev;
+    sha256 = versionsImport.${baseName}.libVersion.libSources.${name}.sha256;
+  };
+
+  # only override `src` or `version` if building `kicad-unstable` with
+  # the appropriate attribute defined in `srcs`.
+  srcOverridep = attr: (!stable && builtins.hasAttr attr srcs);
+
+  # use default source and version (as defined in versions.nix) by
+  # default, or use the appropriate attribute from `srcs` if building
+  # unstable with `srcs` properly defined.
+  kicadSrc =
+    if srcOverridep "kicad" then srcs.kicad
+    else kicadSrcFetch;
+  kicadVersion =
+    if srcOverridep "kicadVersion" then srcs.kicadVersion
+    else versionsImport.${baseName}.kicadVersion.version;
+
+  i18nSrc = if srcOverridep "i18n" then srcs.i18n else i18nSrcFetch;
+  i18nVersion =
+    if srcOverridep "i18nVersion" then srcs.i18nVersion
+    else versionsImport.${baseName}.libVersion.version;
+
+  libSrc = name: if srcOverridep name then srcs.${name} else libSrcFetch name;
+  # TODO does it make sense to only have one version for all libs?
+  libVersion =
+    if srcOverridep "libVersion" then srcs.libVersion
+    else versionsImport.${baseName}.libVersion.version;
+
+  wxGTK =
+    if (stable)
     # wxGTK3x may default to withGtk2 = false, see #73145
-    then wxGTK30.override { withGtk2 = false; }
+    then
+      wxGTK30.override
+        {
+          withGtk2 = false;
+        }
     # wxGTK31 currently introduces an issue with opening the python interpreter in pcbnew
     # but brings high DPI support?
-    else wxGTK31.override { withGtk2 = false; };
+    else
+      wxGTK31.override {
+        withGtk2 = false;
+      };
 
   python = python3;
   wxPython = python.pkgs.wxPython_4_0;
 
 in
 stdenv.mkDerivation rec {
 
-  passthru.libraries = callPackages ./libraries.nix versionConfig.libVersion;
+  passthru.libraries = callPackages ./libraries.nix { inherit libSrc libVersion; };
+  passthru.i18n = callPackage ./i18n.nix {
```

Comment:
OTH, I agree that putting `i18n` outside of libraries makes more sense, since it's a requirement of `base.nix` and not the rest, so... maybe make `passthru.libraries.i18n = throw "this attribute has moved to "kicad-unstable.passthru.i18n"`. Or maybe use `lib.warn`.

Reply:
I'm not sure I completely understand this point. Why do we need to pass i18n to libraries at all? Aren't libraries only called through kicad?

Reply:
It's not a question of whether `libraries.nix` needs `i18n` or not, we know `base.nix` needs it. It's a question of whether we should change the "API" of `kicad-unstable.passsthru`. Someone out there may be relying on it, if they have maybe some other build linking against `kicad-unstable.libraries.i18n`, this PR might tell them to use `passthru.i18n` instead, and I'm just unsure whether this is justified.

Reply:
We are already doing some backwards incompatible changes. Hence I'd keep things as are :). Sorry for maybe misleading you. I'm thinking as I'm reviewing.